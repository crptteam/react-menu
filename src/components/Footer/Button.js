import React from 'react';
import PropTypes from 'prop-types';
import ButtonIcon from '../../svg/buttonIcon.svg';
import ButtonWrapper from '../../styled/ButtonWrapper';

const propTypes = {
  onClick: PropTypes.func,
};

const defaultProps = {
  onClick: () => {},
};

const Button = ({
  onClick,
}) => (
  <ButtonWrapper onClick={onClick}>
    <ButtonIcon />
  </ButtonWrapper>
);

Button.displayName = 'Button';
Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
