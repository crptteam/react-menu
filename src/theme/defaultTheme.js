export default {
  Menu: {
    common: {
      width: {
        full: '241px',
        fold: '78px',
      },
      backgroundColor: '#FFFFFF',
      borderColor: 'rgba(0, 0, 0, 0.25)',
      borderWidth: '1px',
      shadowX: '0px',
      shadowY: '2px',
      shadowColor: 'rgba(0, 0, 0, 0.25)',
    },
    header: {
      height: '200px',
      alignItems: 'center',
      marginTop: '10px',
      marginBottom: '10px',
    },
    item: {
      width: '80%',
      height: '52px',
      borderRadius: '4px',
      marginTop: '10px',
      marginBottom: '10px',
      hoverBorderWidth: '1px',
      hoverBorderColor: '#F8EC31',
      shadowX: '0px',
      shadowY: '2px',
      shadowColor: 'rgba(0, 0, 0, 0.25)',
      justifyContentOnFull: 'flex-start',
      justifyContentOnFold: 'center',
      isSelectedColor: '#F8EC31',
      activeStyle: {
        isSelectedColor: '#F8EC31',
      },
      textColor: 'black',
    },
    fullModeButton: {
      height: '80px',
    },
    separator: {
      width: '80%',
      height: '1px',
      color: 'rgba(0, 0, 0, 0.25)',
    },
    footer: {
      height: '240px',
      alignItems: 'center',
      marginTop: '10px',
      marginBottom: '10px',
    },
  },
};
