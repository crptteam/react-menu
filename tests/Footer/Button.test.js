import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../../src/theme/defaultTheme';

import Button from '../../src/components/Footer/Button';

describe('Footer/Button', () => {
  it('should renders without problem', () => {
    const onClick = jest.fn();
    const component = mount(
      <ThemeProvider theme={defaultTheme}>
        <Button onClick={onClick} />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });

  it('should call onClick', () => {
    const onClick = jest.fn();
    const component = mount(
      <ThemeProvider theme={defaultTheme}>
        <Button onClick={onClick} />
      </ThemeProvider>
    );
    component.find('svg').simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
