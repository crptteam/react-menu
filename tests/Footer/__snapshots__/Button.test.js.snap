// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer/Button should renders without problem 1`] = `
<ThemeProvider
  theme={
    Object {
      "Menu": Object {
        "common": Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "rgba(0, 0, 0, 0.25)",
          "borderWidth": "1px",
          "shadowColor": "rgba(0, 0, 0, 0.25)",
          "shadowX": "0px",
          "shadowY": "2px",
          "width": Object {
            "fold": "78px",
            "full": "241px",
          },
        },
        "footer": Object {
          "alignItems": "center",
          "height": "240px",
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "fullModeButton": Object {
          "height": "80px",
        },
        "header": Object {
          "alignItems": "center",
          "height": "200px",
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "item": Object {
          "borderRadius": "4px",
          "height": "52px",
          "hoverBorderColor": "#F8EC31",
          "hoverBorderWidth": "1px",
          "isSelectedColor": "#F8EC31",
          "justifyContentOnFold": "center",
          "justifyContentOnFull": "flex-start",
          "marginBottom": "10px",
          "marginTop": "10px",
          "shadowColor": "rgba(0, 0, 0, 0.25)",
          "shadowX": "0px",
          "shadowY": "2px",
          "width": "80%",
        },
        "separator": Object {
          "color": "rgba(0, 0, 0, 0.25)",
          "height": "1px",
          "width": "80%",
        },
      },
    }
  }
>
  <Button
    onClick={[MockFunction]}
  >
    <WithTheme(ButtonWrapper)
      onClick={[MockFunction]}
    >
      <ButtonWrapper
        onClick={[MockFunction]}
        theme={
          Object {
            "Menu": Object {
              "common": Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "rgba(0, 0, 0, 0.25)",
                "borderWidth": "1px",
                "shadowColor": "rgba(0, 0, 0, 0.25)",
                "shadowX": "0px",
                "shadowY": "2px",
                "width": Object {
                  "fold": "78px",
                  "full": "241px",
                },
              },
              "footer": Object {
                "alignItems": "center",
                "height": "240px",
                "marginBottom": "10px",
                "marginTop": "10px",
              },
              "fullModeButton": Object {
                "height": "80px",
              },
              "header": Object {
                "alignItems": "center",
                "height": "200px",
                "marginBottom": "10px",
                "marginTop": "10px",
              },
              "item": Object {
                "borderRadius": "4px",
                "height": "52px",
                "hoverBorderColor": "#F8EC31",
                "hoverBorderWidth": "1px",
                "isSelectedColor": "#F8EC31",
                "justifyContentOnFold": "center",
                "justifyContentOnFull": "flex-start",
                "marginBottom": "10px",
                "marginTop": "10px",
                "shadowColor": "rgba(0, 0, 0, 0.25)",
                "shadowX": "0px",
                "shadowY": "2px",
                "width": "80%",
              },
              "separator": Object {
                "color": "rgba(0, 0, 0, 0.25)",
                "height": "1px",
                "width": "80%",
              },
            },
          }
        }
      >
        <ButtonWrapper__ButtonDiv
          fullModeButtonTheme={
            Object {
              "height": "80px",
            }
          }
          onClick={[MockFunction]}
          theme={
            Object {
              "Menu": Object {
                "common": Object {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "rgba(0, 0, 0, 0.25)",
                  "borderWidth": "1px",
                  "shadowColor": "rgba(0, 0, 0, 0.25)",
                  "shadowX": "0px",
                  "shadowY": "2px",
                  "width": Object {
                    "fold": "78px",
                    "full": "241px",
                  },
                },
                "footer": Object {
                  "alignItems": "center",
                  "height": "240px",
                  "marginBottom": "10px",
                  "marginTop": "10px",
                },
                "fullModeButton": Object {
                  "height": "80px",
                },
                "header": Object {
                  "alignItems": "center",
                  "height": "200px",
                  "marginBottom": "10px",
                  "marginTop": "10px",
                },
                "item": Object {
                  "borderRadius": "4px",
                  "height": "52px",
                  "hoverBorderColor": "#F8EC31",
                  "hoverBorderWidth": "1px",
                  "isSelectedColor": "#F8EC31",
                  "justifyContentOnFold": "center",
                  "justifyContentOnFull": "flex-start",
                  "marginBottom": "10px",
                  "marginTop": "10px",
                  "shadowColor": "rgba(0, 0, 0, 0.25)",
                  "shadowX": "0px",
                  "shadowY": "2px",
                  "width": "80%",
                },
                "separator": Object {
                  "color": "rgba(0, 0, 0, 0.25)",
                  "height": "1px",
                  "width": "80%",
                },
              },
            }
          }
        >
          <div
            className="ButtonWrapper__ButtonDiv-hjHEpT efBHuW"
            onClick={[MockFunction]}
          >
            <ButtonIcon
              fill="none"
              height="44"
              viewBox="0 0 44 44"
              width="44"
              xmlns="http://www.w3.org/2000/svg"
            >
              <svg
                fill="none"
                height="44"
                viewBox="0 0 44 44"
                width="44"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M27 23H17c-.6 0-1-.4-1-1s.4-1 1-1h10c.6 0 1 .4 1 1s-.4 1-1 1zM27 19H17c-.6 0-1-.4-1-1s.4-1 1-1h10c.6 0 1 .4 1 1s-.4 1-1 1zM27 27H17c-.6 0-1-.4-1-1s.4-1 1-1h10c.6 0 1 .4 1 1s-.4 1-1 1z"
                  fill="#434244"
                />
                <circle
                  cx="22"
                  cy="22"
                  r="21.5"
                  stroke="#C4C4C4"
                  strokeOpacity=".25"
                />
              </svg>
            </ButtonIcon>
          </div>
        </ButtonWrapper__ButtonDiv>
      </ButtonWrapper>
    </WithTheme(ButtonWrapper)>
  </Button>
</ThemeProvider>
`;
